var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -112,6 +115,8 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     conflicts("+cuda", when="+int64")
     conflicts("+rocm", when="+int64")
     conflicts("+rocm", when="@:2.20")
+    conflicts("+cublas", when="~cuda", msg="cuBLAS requires CUDA to be enabled")
+    conflicts("+rocblas", when="~rocm", msg="rocBLAS requires ROCm to be enabled")
```

Comment:
We should also probably add a conflict for `gpu-aware-mpi` here too.

Reply:
What do you have in mind for the conflict? gpu-aware-mpi works on cuda and rocm, though you need the most recent architectures/devices on each.

Reply:
@PaulMullowney we can add those architectures specifics as requirements. Can you provide what the minimum devices that support this feature are?